aicsimageio/readers/ome_tiff_reader.py

```diff
@@ -0,0 +1,330 @@
+            )
+        # If non global linear timescale, we need to create an array of every plane
+        # time value
+        elif scene_meta.pixels.size_t > 1:
+            t_index_to_delta_map = {
+                p.the_t: p.delta_t for p in scene_meta.pixels.planes
+            }
+            coords[DimensionNames.Time] = list(t_index_to_delta_map.values())
+
+        # Handle Spatial Dimensions
+        if scene_meta.pixels.physical_size_z is not None:
+            coords[DimensionNames.SpatialZ] = np.arange(
+                0,
+                scene_meta.pixels.size_z * scene_meta.pixels.physical_size_z,
+                scene_meta.pixels.physical_size_z,
+            )
+
+        if scene_meta.pixels.physical_size_y is not None:
+            coords[DimensionNames.SpatialY] = np.arange(
+                0,
+                scene_meta.pixels.size_y * scene_meta.pixels.physical_size_y,
+                scene_meta.pixels.physical_size_y,
+            )
+
+        if scene_meta.pixels.physical_size_x is not None:
+            coords[DimensionNames.SpatialX] = np.arange(
+                0,
+                scene_meta.pixels.size_x * scene_meta.pixels.physical_size_x,
+                scene_meta.pixels.physical_size_x,
+            )
+
+        return dims, coords
+
+    @staticmethod
+    def _expand_dims_to_match_ome(
+        image_data: types.ArrayLike,
+        ome: OME,
+        dims: List[str],
+        scene_index: int,
+    ) -> types.ArrayLike:
+        # Expand image_data for empty dimensions
+        ome_shape = []
+        for d in dims:
+            ome_shape.append(
+                getattr(ome.images[scene_index].pixels, f"size_{d.lower()}")
+            )
+
+        # Check for num samples and expand dims if greater than 1
+        n_samples = ome.images[scene_index].pixels.channels[0].samples_per_pixel
+        if n_samples > 1:
+            # Append to the end, i.e. the last dimension
+            dims.append("S")
+            ome_shape.append(n_samples)
+
+        # The file may not have all the data but OME requires certain dimensions
+        # expand to fill
+        expand_dim_ops = []
+        for d_size in ome_shape:
+            # Add empty dimension where OME requires dimension but no data exists
+            if d_size == 1:
+                expand_dim_ops.append(None)
+            # Add full slice where data exists
+            else:
+                expand_dim_ops.append(slice(None, None, None))
+
+        # Apply operators to dask array
+        return image_data[tuple(expand_dim_ops)]
+
+    def _general_data_array_constructor(
+        self, image_data: types.ArrayLike
+    ) -> xr.DataArray:
+        # Get unprocessed metadata from tags
+        tiff_tags = self._get_tiff_tags()
+
+        # Create OME
+        with self.fs.open(self.path) as open_resource:
+            with TiffFile(open_resource) as tiff:
+                ome = self._get_ome(tiff.pages[0].description, self.clean_metadata)
```

Comment:
i see this called in a couple of places -- is it worth caching the result of self._get_ome ? just asking.

Reply:
I thought about this but in each place it is used, it's meant to be a single instance of a place where some sort of object construction or validation occurs.

* `_is_supported_image` is meant to be a bool and we have to run that function _prior_ to setting state because that `_get_ome` itself, can raise an OME validation error which would indicate the file isn't a well structured OME image.
* `.scenes` is the next instance it is used and scene management is the first thing we do after init, because we need to see which scene to actually create the image object for / init state for object.
* `OME` attachment during the actual image array construction to me just makes sense especially because we are passing it through to the xarray attrs to keep code paths consistent.

I guess it's possible that we could cache the `OME` object after `.scenes` is called / created for use in the `xarray` setup but :shrug: its low overhead / quick imo and it would make the code path look a bit weird.

Reply:
After I posted this I looked again. I will cache it during the scene creation and then use the cached during xarray construction.

So it will still need to be separate for the is_supported_image check.